cirq/google/engine/engine.py

```diff
@@ -392,11 +407,53 @@ def run_sweep(
                 }
             },
         }
+        if run_context is not None:
+            request['run_context'] = run_context
         response = self.service.projects().programs().jobs().create(
             parent=response['name'], body=request).execute()
 
         return EngineJob(job_config, response, self)
 
+    def _serialize_program_v1(
+            self, program: Program, sweeps: List[Sweep], repetitions: int
+    ) -> Tuple[Dict[str, Any], Optional[Dict[str, Any]]]:
+        schedule = self.program_as_schedule(program)
+        schedule.device.validate_schedule(schedule)
+
+        descriptor = v1.program_pb2.Program.DESCRIPTOR
+
+        program_dict = {}  # type: Dict[str, Any]
+        program_dict['@type'] = 'type.googleapis.com/' + descriptor.full_name
+        program_dict['parameter_sweeps'] = [
+            sweep_to_proto_dict(sweep, repetitions) for sweep in sweeps
+        ]
+        program_dict['operations'] = [
+            op for op in schedule_to_proto_dicts(schedule)
+        ]
+        return program_dict, None  # run context included in program
```

Comment:
is there a reason to do this for v1 protos?  It would be nice to always just the the run context.

Reply:
We previously talked about making these changes separately, so I left it as-is. It should be straightforward, so I'll address this after attending to the other comments.

Reply:
SGTM.

Reply:
Added. PTAL.